### PR TITLE
Sharing: add Bluesky sharing button

### LINKED ISSUE #32708: Sharing: add Bluesky option
### Impacted plugin

Jetpack

### What

When the new Bluesky platform offers sharing URLs, let's add a Bluesky sharing button to our list of options

### How

See #31537 for prior art (this is the last button we added).

Related: #30460 for Jetpack Social (née Publicize).

### PR DESCRIPTION
**Pull request by jeherve**
> [!NOTE]
> This replaces #36091

Fixes #32708
Closes #36091

## Proposed changes:

This is a replay of #36091 with some fixes, since I couldn't push the fixes to the fork directly.

The PR introduces a new Bluesky sharing button, for both the legacy sharing buttons and the newer sharing buttons block.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* N/A

## Does this pull request change what data or activity we track or use?
* No

## Testing instructions:

**Start by testing the legacy sharing buttons**

1. Go to Appearance > Themes
2. Install and activate the Twenty Ten theme
3. Go to wordpress.com/marketing/sharing-buttons/
4. Add a Bluesky sharing button to your site (note that https://github.com/Automattic/wp-calypso/pull/88180, https://github.com/Automattic/wp-calypso/pull/88202, and https://github.com/Automattic/jetpack/pull/36187 will need to be merged first)
5. After saving, it should look good on your site.
6. Try using different button styles (official, icon, icon+text, text)
7. Next, go to Appearance > Themes and switch to a block-based theme like Twenty Twenty Four
8. Go to Jetpack > Settings > Sharing and disable the sharing feature.
9. Go to Apperance > Site Editor > Templates > Single Posts
10. Add a sharing buttons block to your template, and add a Bluesky button.
11. Ensure it is displayed properly on the frontend, and that the button works well.

### CODE REVIEW

**Review by monsieur-z — DISMISSED** [review]
I'm seeing some discrepancies between the editor and the site.

| Editore | Site |
|-|-|
|Screenshot 2024-03-07 at 10 13 12 AM|Screenshot 2024-03-07 at 10 13 15 AM
|Screenshot 2024-03-07 at 10 13 30 AM|Screenshot 2024-03-07 at 10 13 33 AM|

**Comment by jeherve** [comment]
@monsieur-z Good catch! This is a problem specific to Twenty Ten. I added a fix in #36283.

**Review by monsieur-z — DISMISSED** [review]
The PR tests well, and code changes look sensible.

Nitpicking: in the text-only style, the text is slightly unbalanced because of a lingering margin (tested with the Twenty Ten theme).
Screenshot 2024-03-11 at 11 13 07 AM

**Comment by jeherve** [comment]
> Nitpicking: in the text-only style, the text is slightly unbalanced because of a lingering margin (tested with the Twenty Ten theme).

Good catch. It looks like the margin has been around since the first version of the buttons, in f2b77b308e1df9231914615ab1f333cb3b267ad7.

I think the margin makes sense for the icon + text style, where we have an icon before the text, but it makes less sense for other button styles. I've updated things in b294eaa0b65f5da29017a9d1cebc4ec17cd60f92
Let me know what you think!

File: projects/plugins/jetpack/extensions/blocks/sharing-button/style.scss [hunk 1]
```diff
@@ -82,14 +88,6 @@ a.jetpack-sharing-button__button {
...
 }
 
-.jetpack-sharing-buttons__service-label.style-icon {
-	display: none;
-}
-
-.jetpack-sharing-button__service-label {
-	margin-inline-start: 5px;
-}
-
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/sharing-button/style.scss:92]: This is now only applied to buttons with the icon+text style, see below.

File: projects/plugins/jetpack/extensions/blocks/sharing-button/style.scss [hunk 2]
```diff
@@ -82,14 +88,6 @@ a.jetpack-sharing-button__button {
 	display: none;
 }
 
-.jetpack-sharing-buttons__service-label.style-icon {
-	display: none;
-}
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/sharing-button/style.scss:87]: This didn't seem necessary; that specific selector doesn't exist. I believe the correct one would be `.jetpack-sharing-button__service-label` (without the s), and that style is applied just above.

**Review by monsieur-z — DISMISSED** [review]
Looks good!

**Review by monsieur-z — APPROVED** [review]

**Pull request merged**